include/robot_interfaces/robot_backend.hpp

```diff
@@ -177,11 +201,29 @@ class RobotBackend
     {
         real_time_tools::set_cpu_dma_latency(0);
 
+        const double start_time =
+            real_time_tools::Timer::get_current_time_sec();
+
         // wait until first desired_action was received
         // ----------------------------
         while (!has_shutdown_request() &&
                !robot_data_->desired_action->wait_for_timeindex(0, 0.1))
         {
+            const double now = real_time_tools::Timer::get_current_time_sec();
+            if (now - start_time > first_action_timeout_)
+            {
+                Status status;
+                status.error_status = Status::ErrorStatus::BACKEND_ERROR;
+                status.error_message = "First action was not provided in time";
+
+                robot_data_->status->append(status);
+
+                std::cerr << "Error: " << status.error_message
+                          << "\nRobot is shut down." << std::endl;
+
+                this->shutdown();
```

Comment:
doesn't this wait forever? inside shutdown you join the thread, but here you are inside of that thread.. am i missing something?

Reply:
Ohh.. you are right...  I wonder why this was not happening when I tested.  Will change it in any case.

Reply:
So it does not block but throws an `std::system_error("Resource deadlock avoided")`.  I did not check carefully enough earlier so I did not notice it was not the expected exception.  I should probably add a unit test for this.

Reply:
I replaced it with a non-blocking `request_shutdown` now.